libr/anal/p/anal_xtensa.c

```diff
@@ -613,6 +619,1295 @@ static XtensaOpFn xtensa_op0_fns[] = {
+
+	xtensa_regfile src_rf = xtensa_operand_regfile (isa, opcode, 0);
+	xtensa_regfile bit_rf = xtensa_operand_regfile (isa, opcode, 1);
+
+	// bbc
+	bit_clear = opcode == 70;
+	cmp_op = bit_clear ? "==" : "==,!";
+
+	sign_extend (&imm_offset, 7);
+	imm_offset += 4 - 3;
+
+	// example: bbc a4, a2, offset
+	//          a2,
+	//          1,
+	//          <<,
+	//          a4,
+	//          &
+	//          0
+	//          ==,
+	//          ?{,
+	//            offset,
+	//            pc,
+	//            +=,
+	//          }
+
+	r_strbuf_appendf (
+		&op->esil,
+			"%s%d"	CM
+			"1"	CM
+			"<<"	CM
+			"%s%d"	CM
+			"&"	CM
+			"0"	CM
+			"%s"	CM
+			"?{"	CM,
+		xtensa_regfile_shortname (isa, bit_rf),
+		bit_reg,
+		xtensa_regfile_shortname (isa, src_rf),
+		src_reg,
+		cmp_op
+	);
+
+	esil_push_signed_imm (&op->esil, imm_offset);
+
+	r_strbuf_appendf (
+		&op->esil,
+			"pc"	CM
+			"+="	CM
+			"}"
+	);
+}
+
+static void esil_abs_neg(xtensa_isa isa, xtensa_opcode opcode, xtensa_format format,
+		size_t i, xtensa_insnbuf slot_buffer, RAnalOp *op) {
+	uint32_t src_reg;
+	uint32_t dst_reg;
+
+	uint8_t neg;
+
+	xtensa_operand_get_field (isa, opcode, 1, format, i, slot_buffer, &dst_reg);
+	xtensa_operand_get_field (isa, opcode, 0, format, i, slot_buffer, &src_reg);
+
+	xtensa_regfile dst_rf = xtensa_operand_regfile (isa, opcode, 1);
+	xtensa_regfile src_rf = xtensa_operand_regfile (isa, opcode, 0);
+
+	neg = opcode == 95;
+
+	if (!neg) {
+		r_strbuf_appendf (
+			&op->esil,
+				"0"	CM
+				"%s%d"	CM
+				"<"	CM
+				"?{"	CM
+				"0"     CM
+				"%s%d"	CM
+				"-"     CM
+				"}"	CM
+				"0"	CM
+				"%s%d"	CM
+				">="	CM
+				"?{"	CM
+				"%s%d"	CM
+				"}"	CM,
+			xtensa_regfile_shortname (isa, src_rf),
+			src_reg,
+			xtensa_regfile_shortname (isa, src_rf),
+			src_reg,
+			xtensa_regfile_shortname (isa, src_rf),
+			src_reg,
+			xtensa_regfile_shortname (isa, src_rf),
+			src_reg
+		);
+	} else {
+		r_strbuf_appendf (
+			&op->esil,
+				"0"	CM
+				"%s%d"	CM
+				"-"	CM,
+			xtensa_regfile_shortname (isa, src_rf),
+			src_reg
+		);
+	}
+
+	r_strbuf_appendf (
+		&op->esil,
+			"%s%d"	CM
+			"="	CM,
+		xtensa_regfile_shortname (isa, dst_rf),
+		dst_reg
+	);
+}
+
+static void esil_call(xtensa_isa isa, xtensa_opcode opcode,
+		xtensa_format format, size_t i, xtensa_insnbuf slot_buffer, RAnalOp *op) {
+	uint8_t call;
+	int32_t imm_offset;
+
+	call = opcode == 76;
+
+	xtensa_operand_get_field (isa, opcode, 0, format, i, slot_buffer,
+			(uint32_t *) &imm_offset);
+
+	if (call) {
+		r_strbuf_append(
+			&op->esil,
+			"pc"	CM
+			"a0"	CM
+			"="	CM
+		);
+	}
+
+	sign_extend (&imm_offset, 17);
+
+	if (call) {
+		imm_offset <<= 2;
+	}
+
+	imm_offset += 4 - 3;
+
+	esil_push_signed_imm (&op->esil, imm_offset);
+
+	r_strbuf_append (&op->esil, "pc" CM "+=");
+}
+
+static void esil_callx(xtensa_isa isa, xtensa_opcode opcode,
+		xtensa_format format, size_t i, xtensa_insnbuf slot_buffer, RAnalOp *op) {
+	uint8_t callx;
```

Comment:
use bool and initialize and define in the same line `bool callx = opcode == 77;`